staging/src/k8s.io/client-go/tools/events/helpers.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package events
+
+import (
+	"fmt"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/api/events/v1beta1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// GetFieldSelector returns the appropriate field selector based on the API version being used to communicate with the server.
+// The returned field selector can be used with List and Watch to filter desired events.
+func GetFieldSelector(groupVersion schema.GroupVersion, regardingName, regardingNamespace, regardingKind, regardingUID string) (fields.Selector, error) {
```

Comment:
can you add a test to test/integration/events that creates an event for an object, and queries each of these attributes via the events.k8s.io/v1beta1 and v1 APIs, then each of these attributes with a non-matching value, to make sure these are wired up properly?

Reply:
Sure.